eth/catalyst/api_test.go

```diff
@@ -51,6 +51,10 @@ import (
 	"github.com/mattn/go-colorable"
 )
 
+func init() {
+	miner.IsPayloadBuildingTest = true
```

Comment:
I'm not a fan of the global parameter either. Can we not pass it through a `context` or something, as an optional context value?

Reply:
Given https://github.com/ethereum-optimism/op-geth/blob/c51655402f900cdafdd86f9dab84a13f735f36ca/eth/catalyst/api_test.go#L206 I wonder if these tests were always flaky - a fixed time based wait is never a good sign.

Yep - flaky even without these changes as they may return an empty block and fail. And they use different amounts of time to wait in different tests... 

I had a play with it in https://github.com/ethereum-optimism/op-geth/pull/199 - it's still an arbitrary sleep but it's 1 second.  There's only a couple of tests that need it and so it's better to have a long period and less flakiness.  The tests do have an ethservice which can get to the miner so it may be possible to add an API to check if the full block has been built yet and wait deterministically but I didn't try. That's passed in CI and passed consistently locally.

Reply:
Thanks, merged #199 into this PR.

@protolambda it wasn't possible to pass anything down in tests in general because some tests would just call fcu on the engine API or similar, which don't have a context.

Reply:
Merged in #204 to address this.